R-package/R/lgb.Booster.R

```diff
@@ -231,21 +231,33 @@ Booster <- R6::R6Class(
           private$set_objective_to_none <- TRUE
         }
         # Perform objective calculation
-        gpair <- fobj(private$inner_predict(1L), private$train_set)
+        preds <- private$inner_predict(1L)
+        gpair <- fobj(preds, private$train_set)
 
         # Check for gradient and hessian as list
         if (is.null(gpair$grad) || is.null(gpair$hess)) {
           stop("lgb.Booster.update: custom objective should
             return a list with attributes (hess, grad)")
         }
 
+        # Check grad and hess have the right shape
+        n_grad <- length(gpair$grad)
+        n_hess <- length(gpair$hess)
+        n_preds <- length(preds)
+        if (n_grad != n_preds) {
+          stop(sprintf("grad should have length %d, got %d", n_preds, n_grad))
```

Comment:
```suggestion
          stop(sprintf("Expected custom objective function to return grad with length %d, got %d.", n_preds, n_grad))
```

Can we please modify this error message (and the one about the hessian) to make it clearer to users that their custom objective function's implementation is the most likely cause?